### PR TITLE
Deletion of `AbstractData` 

### PR DESCRIPTION
**Pull request by guimarqu**
Task of #330 

- Storage dict are now stored in reformulation and formulations.
- Deletion of `AbstractData`

PR #509 must be reviewed and merged before this one.

### CODE REVIEW

File: src/Algorithm/basic/solvelpform.jl [hunk 1]
```diff
@@ -64,7 +63,7 @@ function run!(algo::SolveLpForm, env::Env, data::ModelData, input::OptimizationI
     partial_sol = nothing
     partial_sol_val = 0.0
     if algo.consider_partial_solution
-        partsolunit = getunit(data, PartialSolutionUnitPair)
+        partsolunit = getunit(form, PartialSolutionUnitPair)
```
Comment (rrsadykov) [inline src/Algorithm/basic/solvelpform.jl:66]: May be, we rename `getunit` to `getstorageunit` to be more explicit?

File: src/MathProg/formulation.jl [hunk 2]
```diff
@@ -7,6 +7,7 @@ mutable struct Formulation{Duty <: AbstractFormDuty}  <: AbstractFormulation
     manager::FormulationManager
     obj_sense::Type{<:Coluna.AbstractSense}
     buffer::FormulationBuffer
+    storagedict::StorageDict
```
Comment (rrsadykov) [inline src/MathProg/formulation.jl:10]: Just `Storage` instead of `StorageDict`?

Comment (guimarqu) [reply src/MathProg/formulation.jl]: I just renamed the type at the moment.

File: src/ColunaBase/storage.jl [hunk 3]
```diff
@@ -58,7 +58,11 @@ is called during storing a unit.
 abstract type AbstractRecord end
 
 """
+<<<<<<< HEAD
```
Comment (rrsadykov) [inline src/ColunaBase/storage.jl:61]: ???

**Review by rrsadykov — CHANGES_REQUESTED** [review]

File: src/ColunaBase/storage.jl (hunk 3 above)
Comment (guimarqu) [reply src/ColunaBase/storage.jl:61]: Oups

**Review by rrsadykov — APPROVED** [review]

**Pull request merged**